python_modules/libraries/dagster-pyspark/dagster_pyspark/types.py

```diff
@@ -13,17 +23,476 @@
+                            ),
+                            EnumValue('overwrite', description="Overwrite existing data."),
+                            EnumValue(
+                                'ignore',
+                                description="Silently ignore this operation if data already exists.",
+                            ),
+                            EnumValue(
+                                'error',
+                                description="(default case): Throw an exception if data already exists.",
+                            ),
+                            EnumValue(
+                                'errorifexists',
+                                description="(default case): Throw an exception if data already exists.",
+                            ),
+                        ],
+                    ),
+                    is_required=False,
+                    description="specifies the behavior of the save operation when data already exists.",
+                ),
+                'partitionBy': Field(
+                    String, is_required=False, description="names of partitioning columns."
+                ),
+                'options': Field(
+                    Permissive(), is_required=False, description="all other string options."
+                ),
+            },
+            'text': {
+                'path': Field(
+                    String,
+                    is_required=True,
+                    description="he path in any Hadoop supported file system.",
+                ),
+                'compression': Field(
+                    Enum(
+                        'compression',
+                        [
+                            EnumValue('none'),
+                            EnumValue('bzip2'),
+                            EnumValue('gzip'),
+                            EnumValue('lz4'),
+                            EnumValue('snappy'),
+                            EnumValue('deflate'),
+                        ],
+                    ),
+                    is_required=False,
+                    description="compression codec to use when saving to file. This will override ``orc.compress`` and ``spark.sql.orc.compression.codec``. If None is set, it uses the value specified in ``spark.sql.orc.compression.codec``.",
+                ),
+                'lineSep': Field(
+                    String,
+                    is_required=False,
+                    description="defines the line separator that should be used for writing. If None is set, it uses the default value, ``\\n``.",
+                ),
             },
         }
     )
 )
 def spark_df_output_schema(_context, file_type, file_options, spark_df):
     if file_type == 'csv':
         spark_df.write.csv(
-            file_options['path'], header=file_options.get('header'), sep=file_options.get('sep')
+            path=file_options['path'],
```

Comment:
For future proofing, we should probably make the dictionaries of each selector a `Permissive` and then pass in all the kwargs. That way if there is a future upgrade of these APIs the use of them is not contigent on a schema update. Permissive allows you to specify fields that you know about but also to pass in extra ones the schema doesn't know about

Reply:
What about something like this.  
```

Selector(
    {
        'csv': Permissive({
            'path': Field(
                String,
                is_required=True,
                description="the path in any Hadoop supported file system.",
            ),
        })
    }
)

...
```

```
spark_df.write.csv(**file_options)
...
```